### PR TITLE
add flake8-docstrings

### LINKED ISSUE #164: Automatically check for docstrings
We should check that every public interface has a docstring.

https://pypi.org/project/flake8-docstrings/

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/50073634-automatically-check-for-docstrings?utm_campaign=plugin&utm_content=tracker%2F70027116&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F70027116&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #221: remove (or imlement) pkg create's pkglist option
currently `ioc create` has a `--pkglist` option that's not implemented, so we should either implement it, or remove it.

### PR DESCRIPTION
**Pull request by igalic**
- closes #221 (Remove the pkglist option from the CLI create command)
- resolves #164 (Automatically check for docstrings)

### CODE REVIEW

**Comment by igalic** [comment]
i can fix the cosmetic things, but i'll def need help with the substantial ones

**Comment by gronke** [comment]
@igalic I've rebased this branch. The earlier we enable QA for docstrings, the less we have to worry about large documentation sessions.

**Comment by igalic** [comment]
:+1: —  merge whenever you feel ready
mon-wed i might have time for a docs session

**Comment by gronke** [comment]
I preferred refactoring over documenting. We will have to to carefully check the existing features before merging this branch.

File: iocage/lib/Config/Jail/File/RCConf.py [hunk 1]
```diff
@@ -21,166 +21,15 @@
...
 
-class RCConf(
-    dict,
-    iocage.lib.Config.Jail.File.Prototype.ResourceConfigFile
-):
+class RCConf(iocage.lib.Config.Jail.File.Prototype.ResourceConfigFile):
+    """Model a rc.conf file."""
 
-    # the file is always relative to the resource
     _file: str = "/etc/rc.conf"
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/lib/Config/Jail/File/RCConf.py]: Why? It is just very similar to sysctl.conf.

File: iocage/cli/create.py [hunk 2]
```diff
@@ -51,17 +54,14 @@ def validate_count(ctx, param, value):
...
               help="Designate a number of jails to create. Jails are"
                    " numbered sequentially.")
 @click.option("--release", "-r", required=False,
               help="Specify the RELEASE to use for the new jail.")
 @click.option("--template", "-t", required=False,
               help="Specify the template to use for the new jail instead of"
                    " a RELEASE.")
-@click.option("--pkglist", "-p", default=None,
-              help="Specify a JSON file which manages the installation of"
-                   " each package in the newly created jail.")
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/cli/create.py]: Good catch! I've updated the PR body accordingly.

File: iocage/cli/create.py [hunk 3]
```diff
@@ -76,17 +76,25 @@ def validate_count(ctx, param, value):
                    " jails.")
 @click.option("--no-fetch", is_flag=True, default=False,
               help="Do not automatically fetch releases")
-@click.option("--force", "-f", is_flag=True, default=False,
-              help="Skip the interactive question.")
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/cli/create.py]: It was unused and thus should not be reflected in the interface. All accessible options and arguments should be actually implemented.

File: iocage/cli/list.py [hunk 4]
```diff
@@ -51,39 +52,33 @@
...
-            available_releases = host.distribution.releases
-            for available_release in available_releases:
-                logger.screen(available_release.name)
-            return
-        except iocage.lib.errors.IocageException:
-            exit(1)
-
-    if plugins and remote:
-        raise iocage.lib.errors.MissingFeature("Plugins", plural=True)
-
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/cli/list.py]: Work on plugins (or whatever they will become) will be done in a different PR. We don't need fragments of unimplemented features. 😏

File: iocage/lib/Distribution.py [hunk 5]
```diff
@@ -33,6 +34,7 @@
 
 
 class EOLParser(html.parser.HTMLParser):
+    """Parser for EOL releases."""
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/lib/Distribution.py]: Not thought of it yet. EOL warnings are just a notification, but not enforced. If you find it important to support that, feel free to create an enhancement issue.

File: iocage/lib/Jail.py [hunk 6]
```diff
@@ -1388,9 +1344,7 @@ def jid(self) -> typing.Optional[int]:
 
     @property
     def env(self):
-        """
-        Environment variables for hook scripts
-        """
+        """Return the environment variables for hook scripts."""
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/lib/Jail.py]: Neat, isn't it?

File: iocage/lib/DevfsRules.py [hunk 7]
```diff
@@ -238,20 +203,18 @@ def append(self, ruleset, is_system_rule=False):
 
     def new_ruleset(self, ruleset):
         """
-        Append a new ruleset
+        Append a new ruleset.
 
         Similar to append(), but automatically assigns a new number
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/lib/DevfsRules.py]: Wow, I wonder why flake8-docstring didn't complain!

[the remark this answers is not in the document]
Comment (gronke) [reply iocage/lib/DevfsRules.py]: It was just added.

**Comment by gronke** [comment]
Thanks to @yonk42 and @igalic for testing and rewiewing 💜

**Pull request merged**